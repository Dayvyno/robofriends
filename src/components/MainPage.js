import React from 'react';
import CardList from '../components/cardList';
import SearchBox from '../components/searchBox';
import './mainPage.css';
import Scroll from '../components/scroll';
import ErrorBoundry from '../components/errorBoundry';
import Header from '../components/hearder';

class MainPage extends React.Component{

  componentDidMount(){
    this.props.onRequestRobots()
  }

  filteredRobots = ()=>{
    return this.props.robots.filter(robot=>{
     return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
   })
  }


  render(){
    const {onSearchChange, isPending} = this.props
  
    // const filteredRobots = robots.filter(robot=>{
    //   return robot.name.toLowerCase().includes(searchField.toLowerCase())
    // })
  
    return(
      <div>
        {
        isPending? <h1>Loading....</h1> :
        <div style={{padding:"5px"}} >
          <Header/>
          <SearchBox searchHandler={onSearchChange}/>
          <Scroll >
            <ErrorBoundry>
              <CardList robots={this.filteredRobots()} />
            </ErrorBoundry>
          </Scroll>
        </div>
        }
      </div>
    )
  }
}


export default MainPage