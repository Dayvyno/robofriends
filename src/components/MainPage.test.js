import {shallow} from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;
beforeEach(()=>{
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false,
  }
  wrapper = shallow(<MainPage {...mockProps} />)
})

it("It renders mainpage without crashing", ()=>{
  expect(wrapper).toMatchSnapshot()
})

it("It filters robots correctly", ()=>{
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [{
      name: 'Johna',
      id: 3,
      mail: "john@gmail.com"
    }],
    searchField: 'a',
    isPending: false,
  }
  const wrapper2 = shallow(<MainPage {...mockProps2} />)
  // expect(wrapper2.instance().filteredRobots()).toEqual([])
  expect(wrapper2.instance().filteredRobots()).toEqual([
    {
      name: 'Johna',
      id: 3,
      mail: "john@gmail.com"
    }
  ])
})

it("It filters robots correctly2", ()=>{
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [{
      name: 'John',
      id: 3,
      mail: "john@gmail.com"
    }],
    searchField: 'a',
    isPending: false,
  }
  const filteredRobot = []
  const wrapper3 = shallow(<MainPage {...mockProps3} />)
  // expect(wrapper2.instance().filteredRobots()).toEqual([])
  expect(wrapper3.instance().filteredRobots()).toEqual(filteredRobot)
})