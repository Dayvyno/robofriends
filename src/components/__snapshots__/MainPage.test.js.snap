// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders mainpage without crashing 1`] = `
<div>
  <div
    style={
      Object {
        "padding": "5px",
      }
    }
  >
    <Header />
    <SearchBox />
    <Scroll>
      <ErrorBoundry>
        <CardList
          robots={Array []}
        />
      </ErrorBoundry>
    </Scroll>
  </div>
</div>
`;
